### DISCUSSION TITLE
Divergence of vector AuxVariable in AuxKernel

### DISCUSSION THREAD

**Discussion by RossMech** [idaholab/moose#20554] [Q&A General] [answered]
Hello everyone,

I would like to ask if one can calculate divirgence of vector AuxVariable as scalar AuxVariable with some kind of AuxKernel? If there is no such functionality implemented, then how it is possible to call divergence operator inside of AuxKernel source code?

Thank you for the answer in advance!

Best regards,
Rostyslav Nizinkovskyi.

**Comment by GiudGiud** [comment, marked as answer]
Hello

There is this one:
https://mooseframework.inl.gov/source/auxkernels/INSDivergenceAux.html
in the Navier Stokes module. What modules are you currently using?

Nevermind the description in the documentation, it does compute the divergence.
I ll fix the docs in an upcoming PR

Guillaume

**Reply by RossMech** [reply]
Thank you for the answer. I will try to implement it in the model!
